flink-core/src/main/java/org/apache/flink/configuration/StructuredOptionsSplitter.java

```diff
@@ -52,6 +59,37 @@
 		return processTokens(tokens);
 	}
 
+	/**
+	 * When write a string out, it is always needs to escaped. If the input string contains double quote or specified
+	 * escape chars, then it will be escaped by single quote. The single quote will be escaped by doubling.
+	 *
+	 * <p>Given that the escapeChar is (;)
+	 *
+	 * <p>Examples:
+	 * <ul>
+	 *     <li>A,B,C,D => A,B,C,D</li>
+	 *     <li>A;BCD => 'A;BCD'</li>
+	 *     <li>AB"C"D => 'AB"C"D'</li>
+	 *     <li>AB'"D:B => 'AB''"D:B'</li>
+	 * </ul>
+	 *
+	 * @param value a string which needs to be escaped
+	 * @param escapeChars escape chars for the escape conditions
+	 * @return escaped string with single quote
+	 */
+	static String escapeWithSingleQuote(String value, String... escapeChars) {
```

Comment:
I don't necessarily understand why do we need that complex logic. 
Can't we just do `"'" + value.replaceAll("'", "''") + "'"` ?

Moreover we should try not to use any external dependencies. Especially guava.

Reply:
How about:
```
	/**
	 * Escapes the given string with single quotes, if the input string contains a double quote or any of the
	 * given {@code charsToEscape}. Any single quotes in the input string will be escaped by doubling.
	 *
	 * <p>Given that the escapeChar is (;)
	 *
	 * <p>Examples:
	 * <ul>
	 *     <li>A,B,C,D => A,B,C,D</li>
	 *     <li>A;BCD => 'A;BCD'</li>
	 *     <li>AB"C"D => 'AB"C"D'</li>
	 *     <li>AB'"D:B => 'AB''"D:B'</li>
	 * </ul>
	 *
	 * @param string a string which needs to be escaped
	 * @param charsToEscape escape chars for the escape conditions
	 * @return escaped string by single quote
	 */
	static String escapeWithSingleQuote(String string, String... charsToEscape) {
		boolean escape = Arrays.stream(charsToEscape).anyMatch(string::contains) || string.contains("\"");

		if (escape) {
			return "'" + string.replaceAll("'", "''") + "'";
		}

		return string;
	}
```

Reply:
Usually the guava implementation has better performance than java regex replacement. However, i am not against with removing the guava dependencies. I will use `"'" + value.replaceAll("'", "''") + "'"` instead.

Reply:
The escape condition also needs to add `string.contains("'")`.